stdlib/list.ml

```diff
@@ -197,16 +197,26 @@ let partition p l =
   | x :: l -> if p x then part (x :: yes) no l else part yes (x :: no) l in
   part [] [] l
 
-let rec split = function
-    [] -> ([], [])
-  | (x,y)::l ->
-      let (rx, ry) = split l in (x::rx, y::ry)
+let rev_split xys =
+  let rec _rev_split xs ys = function
+    | [] -> xs,ys
+    | (x,y) :: xys -> _rev_split (x::xs) (y::ys) xys
+  in
+  _rev_split [] [] xys
+
+let split xys = rev_split (rev xys)
```

Comment:
Since `rev_split` is not used otherwise, you could just have the internal helper function in `split` and apply `rev` to the initial call.